compute/openstack.py

```diff
@@ -250,6 +260,85 @@ def get_private_ip(self) -> str:
         """Return the private IP address of the VM."""
         return self.node.private_ips[0] if self.node else ""
 
+    def shutdown(self) -> None:
+        """
+        Gracefully power off the VM associated with this instance.
+        """
+        try:
+            LOG.info(f"Initiating shutdown of node: {self.node.name}")
+            self.driver.ex_stop_node(self.node)
+            self._wait_until_vm_shutdown()
```

Comment:
Try to have this as a non-blocking call... otherwise we will be sequential here.